sql/stats: TestCreateStatsControlJob failed

[(sql/stats).TestCreateStatsControlJob failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1717839&tab=buildLog) on [master@730b02b91a94df028abca8e68923e8bf1d80afa6](https://github.com/cockroachdb/cockroach/commits/730b02b91a94df028abca8e68923e8bf1d80afa6):

```
I200130 03:25:15.878683 46581 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200130 03:25:15.881606 46581 storage/store_snapshot.go:973  [n1,replicate,s1,r12/1:/Table/1{6-7}] sending LEARNER snapshot 1f383b71 at applied index 15
I200130 03:25:15.881731 46581 storage/store_snapshot.go:1016  [n1,replicate,s1,r12/1:/Table/1{6-7}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I200130 03:25:15.882364 49874 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r12/1:/Table/1{6-7}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200130 03:25:15.885917 49872 storage/replica_raftstorage.go:786  [n2,s2,r12/2:{-}] applying LEARNER snapshot [id=1f383b71 index=15]
I200130 03:25:15.888488 49872 storage/replica_raftstorage.go:807  [n2,s2,r12/2:/Table/1{6-7}] applied LEARNER snapshot [total=3ms ingestion=4@2ms id=1f383b71 index=15]
I200130 03:25:15.889862 46581 storage/replica_command.go:1706  [n1,replicate,s1,r12/1:/Table/1{6-7}] change replicas (add [(n2,s2):2] remove []): existing descriptor r12:/Table/1{6-7} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200130 03:25:15.894309 46581 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200130 03:25:15.894818 46581 storage/replica_command.go:1706  [n1,replicate,s1,r12/1:/Table/1{6-7}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r12:/Table/1{6-7} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200130 03:25:15.902816 47422 gossip/gossip.go:1532  [n2] node has connected to cluster via gossip
I200130 03:25:15.902964 47422 storage/stores.go:259  [n2] wrote 2 node addresses to persistent storage
I200130 03:25:15.904546 46581 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200130 03:25:15.906825 49901 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r12/1:/Table/1{6-7}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200130 03:25:15.907045 46581 storage/store_snapshot.go:973  [n1,replicate,s1,r12/1:/Table/1{6-7}] sending LEARNER snapshot c9710156 at applied index 19
I200130 03:25:15.907371 46581 storage/store_snapshot.go:1016  [n1,replicate,s1,r12/1:/Table/1{6-7}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 10, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I200130 03:25:15.909710 49911 storage/replica_raftstorage.go:786  [n3,s3,r12/3:{-}] applying LEARNER snapshot [id=c9710156 index=19]
I200130 03:25:15.913412 49911 storage/replica_raftstorage.go:807  [n3,s3,r12/3:/Table/1{6-7}] applied LEARNER snapshot [total=4ms ingestion=4@3ms id=c9710156 index=19]
I200130 03:25:15.922641 46581 storage/replica_command.go:1706  [n1,replicate,s1,r12/1:/Table/1{6-7}] change replicas (add [(n3,s3):3] remove []): existing descriptor r12:/Table/1{6-7} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200130 03:25:15.927172 46581 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200130 03:25:15.929669 46581 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n3,s3):2LEARNER] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, next=2, gen=0]
I200130 03:25:15.933239 46581 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(l2) ADD_REPLICA[(n3,s3):2LEARNER]: after=[(n1,s1):1 (n3,s3):2LEARNER] next=3
I200130 03:25:15.937257 46581 storage/store_snapshot.go:973  [n1,replicate,s1,r14/1:/Table/1{8-9}] sending LEARNER snapshot 9f3e8b31 at applied index 15
I200130 03:25:15.937389 46581 storage/store_snapshot.go:1016  [n1,replicate,s1,r14/1:/Table/1{8-9}] streamed snapshot to (n3,s3):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I200130 03:25:15.937683 49822 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r14/1:/Table/1{8-9}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):2LEARNER
I200130 03:25:15.938614 49948 storage/replica_raftstorage.go:786  [n3,s3,r14/2:{-}] applying LEARNER snapshot [id=9f3e8b31 index=15]
I200130 03:25:15.946574 49948 storage/replica_raftstorage.go:807  [n3,s3,r14/2:/Table/1{8-9}] applied LEARNER snapshot [total=8ms ingestion=4@8ms id=9f3e8b31 index=15]
I200130 03:25:15.947384 46581 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n3,s3):2] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, (n3,s3):2LEARNER, next=3, gen=1]
I200130 03:25:15.953074 46581 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(v2) ADD_REPLICA[(n3,s3):2]: after=[(n1,s1):1 (n3,s3):2] next=3
I200130 03:25:15.953641 46581 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n2,s2):3LEARNER] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, (n3,s3):2, next=3, gen=2]
I200130 03:25:15.968275 46581 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(l3) ADD_REPLICA[(n2,s2):3LEARNER]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3LEARNER] next=4
I200130 03:25:15.970244 49855 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r14/1:/Table/1{8-9}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):3LEARNER
I200130 03:25:15.970393 46581 storage/store_snapshot.go:973  [n1,replicate,s1,r14/1:/Table/1{8-9}] sending LEARNER snapshot 3d8179ea at applied index 20
I200130 03:25:15.970507 46581 storage/store_snapshot.go:1016  [n1,replicate,s1,r14/1:/Table/1{8-9}] streamed snapshot to (n2,s2):3LEARNER: kv pairs: 10, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I200130 03:25:15.971181 49828 storage/replica_raftstorage.go:786  [n2,s2,r14/3:{-}] applying LEARNER snapshot [id=3d8179ea index=20]
I200130 03:25:15.978133 49828 storage/replica_raftstorage.go:807  [n2,s2,r14/3:/Table/1{8-9}] applied LEARNER snapshot [total=7ms ingestion=4@7ms id=3d8179ea index=20]
I200130 03:25:15.979492 46581 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n2,s2):3] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, (n3,s3):2, (n2,s2):3LEARNER, next=4, gen=3]
I200130 03:25:15.982807 46581 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(v3) ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I200130 03:25:15.996041 46581 testutils/testcluster/testcluster.go:760  WaitForFullReplication took: 1.79272269s
I200130 03:25:16.037758 46695 gossip/gossip.go:1532  [n1] node has connected to cluster via gossip
I200130 03:25:16.037980 46695 storage/stores.go:259  [n1] wrote 2 node addresses to persistent storage
I200130 03:25:16.046534 49978 sql/event_log.go:132  [n1,client=127.0.0.1:45564,user=root] Event: "create_database", target: 52, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I200130 03:25:16.060749 49978 sql/event_log.go:132  [n1,client=127.0.0.1:45564,user=root] Event: "create_table", target: 53, info: {TableName:d.public.t Statement:CREATE TABLE d.public.t (x INT8 PRIMARY KEY) User:root}
I200130 03:25:16.063986 50240 storage/replica_command.go:406  [n1,split,s1,r28/1:/{Table/32-Max}] initiating a split of this range at key /Table/53 [r29] (zone config)
--- FAIL: TestCreateStatsControlJob (3.05s)
=== RUN   TestCreateStatsControlJob/cancel
I200130 03:25:16.130877 50370 jobs/registry.go:640  [n1] job 525219647016534017: stepping through state running with error <nil>
E200130 03:25:16.178102 50370 jobs/registry.go:770  [n1] job 525219647016534017: adoption completed with error job 525219647016534017: node liveness error: restarting in background
E200130 03:25:16.178146 50370 jobs/registry.go:774  [n1] job 525219647016534017: failed querying status: context canceled
    --- FAIL: TestCreateStatsControlJob/cancel (0.07s)
        create_stats_job_test.go:82: expected 'cancel-requested' error, but got pq: job 525219647016534017: node liveness error: restarting in background

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCreateStatsControlJob PKG=./pkg/sql/stats TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCreateStatsControlJob.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/stats).TestCreateStatsControlJob failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1717841&tab=buildLog) on [master@730b02b91a94df028abca8e68923e8bf1d80afa6](https://github.com/cockroachdb/cockroach/commits/730b02b91a94df028abca8e68923e8bf1d80afa6):

```
I200130 03:57:16.400324 47694 storage/replica_raft.go:248  [n1,s1,r5/1:/{Systemtse-Table/System…}] proposing SIMPLE(v3) ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I200130 03:57:16.428123 47694 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, next=2, gen=0]
I200130 03:57:16.503290 47694 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200130 03:57:16.521096 47694 storage/store_snapshot.go:973  [n1,replicate,s1,r14/1:/Table/1{8-9}] sending LEARNER snapshot 96fe5f19 at applied index 16
I200130 03:57:16.522107 47694 storage/store_snapshot.go:1016  [n1,replicate,s1,r14/1:/Table/1{8-9}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200130 03:57:16.525839 51827 storage/replica_raftstorage.go:786  [n2,s2,r14/2:{-}] applying LEARNER snapshot [id=96fe5f19 index=16]
I200130 03:57:16.543432 51827 storage/replica_raftstorage.go:807  [n2,s2,r14/2:/Table/1{8-9}] applied LEARNER snapshot [total=17ms ingestion=4@15ms id=96fe5f19 index=16]
I200130 03:57:16.554286 47694 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n2,s2):2] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200130 03:57:16.555788 51811 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r14/1:/Table/1{8-9}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200130 03:57:16.594763 48796 server/status/runtime.go:498  [n2] runtime stats: 908 MiB RSS, 702 goroutines, 64 MiB/102 MiB/118 MiB GO alloc/idle/total, 85 MiB/116 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (85x), 2.6 MiB/2.6 MiB (r/w)net
I200130 03:57:16.611717 47694 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200130 03:57:16.655325 47694 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200130 03:57:16.820067 47694 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200130 03:57:16.835882 47694 storage/store_snapshot.go:973  [n1,replicate,s1,r14/1:/Table/1{8-9}] sending LEARNER snapshot e2571a66 at applied index 20
I200130 03:57:16.836905 47694 storage/store_snapshot.go:1016  [n1,replicate,s1,r14/1:/Table/1{8-9}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 10, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I200130 03:57:16.837277 51898 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r14/1:/Table/1{8-9}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200130 03:57:16.840623 51918 storage/replica_raftstorage.go:786  [n3,s3,r14/3:{-}] applying LEARNER snapshot [id=e2571a66 index=20]
I200130 03:57:16.856766 51918 storage/replica_raftstorage.go:807  [n3,s3,r14/3:/Table/1{8-9}] applied LEARNER snapshot [total=14ms ingestion=4@11ms id=e2571a66 index=20]
I200130 03:57:16.863437 47694 storage/replica_command.go:1706  [n1,replicate,s1,r14/1:/Table/1{8-9}] change replicas (add [(n3,s3):3] remove []): existing descriptor r14:/Table/1{8-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200130 03:57:16.892744 47694 storage/replica_raft.go:248  [n1,s1,r14/1:/Table/1{8-9}] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200130 03:57:16.927185 47694 storage/replica_command.go:1706  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] change replicas (add [(n3,s3):2LEARNER] remove []): existing descriptor r25:/{Table/29-NamespaceTable/30} [(n1,s1):1, next=2, gen=0]
I200130 03:57:17.014382 47694 storage/replica_raft.go:248  [n1,s1,r25/1:/{Table/29-NamespaceTab…}] proposing SIMPLE(l2) ADD_REPLICA[(n3,s3):2LEARNER]: after=[(n1,s1):1 (n3,s3):2LEARNER] next=3
I200130 03:57:17.028755 47694 storage/store_snapshot.go:973  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] sending LEARNER snapshot 930e9ada at applied index 15
I200130 03:57:17.029676 47694 storage/store_snapshot.go:1016  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] streamed snapshot to (n3,s3):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200130 03:57:17.033119 52005 storage/replica_raftstorage.go:786  [n3,s3,r25/2:{-}] applying LEARNER snapshot [id=930e9ada index=15]
I200130 03:57:17.040142 51924 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r25/1:/{Table/29-NamespaceTab…}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):2LEARNER
I200130 03:57:17.044856 52005 storage/replica_raftstorage.go:807  [n3,s3,r25/2:/{Table/29-NamespaceTab…}] applied LEARNER snapshot [total=11ms ingestion=4@9ms id=930e9ada index=15]
I200130 03:57:17.093825 47694 storage/replica_command.go:1706  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] change replicas (add [(n3,s3):2] remove []): existing descriptor r25:/{Table/29-NamespaceTable/30} [(n1,s1):1, (n3,s3):2LEARNER, next=3, gen=1]
I200130 03:57:17.142208 49122 server/status/runtime.go:498  [n3] runtime stats: 919 MiB RSS, 703 goroutines, 69 MiB/96 MiB/121 MiB GO alloc/idle/total, 85 MiB/117 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (86x), 2.7 MiB/2.7 MiB (r/w)net
I200130 03:57:17.187359 47694 storage/replica_raft.go:248  [n1,s1,r25/1:/{Table/29-NamespaceTab…}] proposing SIMPLE(v2) ADD_REPLICA[(n3,s3):2]: after=[(n1,s1):1 (n3,s3):2] next=3
I200130 03:57:17.195959 47694 storage/replica_command.go:1706  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] change replicas (add [(n2,s2):3LEARNER] remove []): existing descriptor r25:/{Table/29-NamespaceTable/30} [(n1,s1):1, (n3,s3):2, next=3, gen=2]
I200130 03:57:17.279454 47694 storage/replica_raft.go:248  [n1,s1,r25/1:/{Table/29-NamespaceTab…}] proposing SIMPLE(l3) ADD_REPLICA[(n2,s2):3LEARNER]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3LEARNER] next=4
I200130 03:57:17.354750 47694 storage/store_snapshot.go:973  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] sending LEARNER snapshot 2a7bf666 at applied index 20
I200130 03:57:17.358950 47694 storage/store_snapshot.go:1016  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] streamed snapshot to (n2,s2):3LEARNER: kv pairs: 10, log entries: 0, rate-limit: 8.0 MiB/sec, 0.07s
I200130 03:57:17.373159 51962 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r25/1:/{Table/29-NamespaceTab…}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):3LEARNER
I200130 03:57:17.395631 52053 storage/replica_raftstorage.go:786  [n2,s2,r25/3:{-}] applying LEARNER snapshot [id=2a7bf666 index=20]
I200130 03:57:17.416255 52053 storage/replica_raftstorage.go:807  [n2,s2,r25/3:/{Table/29-NamespaceTab…}] applied LEARNER snapshot [total=19ms ingestion=4@16ms id=2a7bf666 index=20]
I200130 03:57:17.421670 47694 storage/replica_command.go:1706  [n1,replicate,s1,r25/1:/{Table/29-NamespaceTab…}] change replicas (add [(n2,s2):3] remove []): existing descriptor r25:/{Table/29-NamespaceTable/30} [(n1,s1):1, (n3,s3):2, (n2,s2):3LEARNER, next=4, gen=3]
I200130 03:57:17.466935 47694 storage/replica_raft.go:248  [n1,s1,r25/1:/{Table/29-NamespaceTab…}] proposing SIMPLE(v3) ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I200130 03:57:17.510612 47694 testutils/testcluster/testcluster.go:760  WaitForFullReplication took: 10.158417534s
I200130 03:57:17.750118 51892 sql/event_log.go:132  [n1,client=127.0.0.1:34460,user=root] Event: "create_database", target: 52, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I200130 03:57:18.012971 51892 sql/event_log.go:132  [n1,client=127.0.0.1:34460,user=root] Event: "create_table", target: 53, info: {TableName:d.public.t Statement:CREATE TABLE d.public.t (x INT8 PRIMARY KEY) User:root}
I200130 03:57:18.056500 52361 storage/replica_command.go:406  [n1,split,s1,r28/1:/{Table/32-Max}] initiating a split of this range at key /Table/53 [r29] (zone config)
--- FAIL: TestCreateStatsControlJob (15.72s)
=== RUN   TestCreateStatsControlJob/cancel
I200130 03:57:19.073769 52696 jobs/registry.go:640  [n1] job 525225947717435393: stepping through state running with error <nil>
E200130 03:57:19.330163 52696 jobs/registry.go:770  [n1] job 525225947717435393: adoption completed with error job 525225947717435393: node liveness error: restarting in background
E200130 03:57:19.331762 52696 jobs/registry.go:774  [n1] job 525225947717435393: failed querying status: context canceled
    --- FAIL: TestCreateStatsControlJob/cancel (0.42s)
        create_stats_job_test.go:82: expected 'cancel-requested' error, but got pq: job 525225947717435393: node liveness error: restarting in background

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCreateStatsControlJob PKG=./pkg/sql/stats TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCreateStatsControlJob.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/stats).TestCreateStatsControlJob failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1717904&tab=buildLog) on [master@730b02b91a94df028abca8e68923e8bf1d80afa6](https://github.com/cockroachdb/cockroach/commits/730b02b91a94df028abca8e68923e8bf1d80afa6):

```
I200130 07:02:42.820402 47640 storage/replica_command.go:1706  [n1,replicate,s1,r13/1:/Table/1{7-8}] change replicas (add [(n3,s3):3] remove []): existing descriptor r13:/Table/1{7-8} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200130 07:02:42.913814 47640 storage/replica_raft.go:248  [n1,s1,r13/1:/Table/1{7-8}] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200130 07:02:42.943012 47640 storage/replica_command.go:1706  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] change replicas (add [(n3,s3):2LEARNER] remove []): existing descriptor r2:/System/NodeLiveness{-Max} [(n1,s1):1, next=2, gen=0]
I200130 07:02:43.010613 47640 storage/replica_raft.go:248  [n1,s1,r2/1:/System/NodeLiveness{-Max}] proposing SIMPLE(l2) ADD_REPLICA[(n3,s3):2LEARNER]: after=[(n1,s1):1 (n3,s3):2LEARNER] next=3
I200130 07:02:43.027626 52327 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r2/1:/System/NodeLiveness{-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):2LEARNER
I200130 07:02:43.034948 47640 storage/store_snapshot.go:973  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] sending LEARNER snapshot 5900a54f at applied index 44
I200130 07:02:43.059257 47640 storage/store_snapshot.go:1016  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] streamed snapshot to (n3,s3):2LEARNER: kv pairs: 18, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200130 07:02:43.088719 52374 storage/replica_raftstorage.go:786  [n3,s3,r2/2:{-}] applying LEARNER snapshot [id=5900a54f index=44]
I200130 07:02:43.123744 52374 storage/replica_raftstorage.go:807  [n3,s3,r2/2:/System/NodeLiveness{-Max}] applied LEARNER snapshot [total=35ms ingestion=4@28ms id=5900a54f index=44]
I200130 07:02:43.138131 47640 storage/replica_command.go:1706  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] change replicas (add [(n3,s3):2] remove []): existing descriptor r2:/System/NodeLiveness{-Max} [(n1,s1):1, (n3,s3):2LEARNER, next=3, gen=1]
I200130 07:02:43.201286 47640 storage/replica_raft.go:248  [n1,s1,r2/1:/System/NodeLiveness{-Max}] proposing SIMPLE(v2) ADD_REPLICA[(n3,s3):2]: after=[(n1,s1):1 (n3,s3):2] next=3
I200130 07:02:43.215035 47640 storage/replica_command.go:1706  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] change replicas (add [(n2,s2):3LEARNER] remove []): existing descriptor r2:/System/NodeLiveness{-Max} [(n1,s1):1, (n3,s3):2, next=3, gen=2]
I200130 07:02:43.278459 47640 storage/replica_raft.go:248  [n1,s1,r2/1:/System/NodeLiveness{-Max}] proposing SIMPLE(l3) ADD_REPLICA[(n2,s2):3LEARNER]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3LEARNER] next=4
I200130 07:02:43.302318 47640 storage/store_snapshot.go:973  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] sending LEARNER snapshot 3dfc97d2 at applied index 48
I200130 07:02:43.310604 47640 storage/store_snapshot.go:1016  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] streamed snapshot to (n2,s2):3LEARNER: kv pairs: 22, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200130 07:02:43.311432 52266 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r2/1:/System/NodeLiveness{-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):3LEARNER
I200130 07:02:43.319608 52352 storage/replica_raftstorage.go:786  [n2,s2,r2/3:{-}] applying LEARNER snapshot [id=3dfc97d2 index=48]
I200130 07:02:43.340720 52352 storage/replica_raftstorage.go:807  [n2,s2,r2/3:/System/NodeLiveness{-Max}] applied LEARNER snapshot [total=19ms ingestion=4@8ms id=3dfc97d2 index=48]
I200130 07:02:43.349437 47640 storage/replica_command.go:1706  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] change replicas (add [(n2,s2):3] remove []): existing descriptor r2:/System/NodeLiveness{-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3LEARNER, next=4, gen=3]
I200130 07:02:43.423328 47640 storage/replica_raft.go:248  [n1,s1,r2/1:/System/NodeLiveness{-Max}] proposing SIMPLE(v3) ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I200130 07:02:43.458355 47640 storage/replica_command.go:1706  [n1,replicate,s1,r28/1:/{Table/32-Max}] change replicas (add [(n3,s3):2LEARNER] remove []): existing descriptor r28:/{Table/32-Max} [(n1,s1):1, next=2, gen=0]
I200130 07:02:43.620857 47640 storage/replica_raft.go:248  [n1,s1,r28/1:/{Table/32-Max}] proposing SIMPLE(l2) ADD_REPLICA[(n3,s3):2LEARNER]: after=[(n1,s1):1 (n3,s3):2LEARNER] next=3
I200130 07:02:43.676383 47640 storage/store_snapshot.go:973  [n1,replicate,s1,r28/1:/{Table/32-Max}] sending LEARNER snapshot 9e8bf82b at applied index 18
I200130 07:02:43.678894 52423 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r28/1:/{Table/32-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):2LEARNER
I200130 07:02:43.680425 47640 storage/store_snapshot.go:1016  [n1,replicate,s1,r28/1:/{Table/32-Max}] streamed snapshot to (n3,s3):2LEARNER: kv pairs: 7, log entries: 0, rate-limit: 8.0 MiB/sec, 0.05s
I200130 07:02:43.692107 52370 storage/replica_raftstorage.go:786  [n3,s3,r28/2:{-}] applying LEARNER snapshot [id=9e8bf82b index=18]
I200130 07:02:43.720971 52370 storage/replica_raftstorage.go:807  [n3,s3,r28/2:/{Table/32-Max}] applied LEARNER snapshot [total=28ms ingestion=4@14ms id=9e8bf82b index=18]
I200130 07:02:43.759864 47640 storage/replica_command.go:1706  [n1,replicate,s1,r28/1:/{Table/32-Max}] change replicas (add [(n3,s3):2] remove []): existing descriptor r28:/{Table/32-Max} [(n1,s1):1, (n3,s3):2LEARNER, next=3, gen=1]
I200130 07:02:43.843980 47640 storage/replica_raft.go:248  [n1,s1,r28/1:/{Table/32-Max}] proposing SIMPLE(v2) ADD_REPLICA[(n3,s3):2]: after=[(n1,s1):1 (n3,s3):2] next=3
I200130 07:02:43.850879 47640 storage/replica_command.go:1706  [n1,replicate,s1,r28/1:/{Table/32-Max}] change replicas (add [(n2,s2):3LEARNER] remove []): existing descriptor r28:/{Table/32-Max} [(n1,s1):1, (n3,s3):2, next=3, gen=2]
I200130 07:02:43.945330 47640 storage/replica_raft.go:248  [n1,s1,r28/1:/{Table/32-Max}] proposing SIMPLE(l3) ADD_REPLICA[(n2,s2):3LEARNER]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3LEARNER] next=4
I200130 07:02:43.995780 47640 storage/store_snapshot.go:973  [n1,replicate,s1,r28/1:/{Table/32-Max}] sending LEARNER snapshot 40999e2c at applied index 24
I200130 07:02:43.996842 52549 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r28/1:/{Table/32-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):3LEARNER
I200130 07:02:44.018078 47640 storage/store_snapshot.go:1016  [n1,replicate,s1,r28/1:/{Table/32-Max}] streamed snapshot to (n2,s2):3LEARNER: kv pairs: 11, log entries: 0, rate-limit: 8.0 MiB/sec, 0.04s
I200130 07:02:44.023398 52448 storage/replica_raftstorage.go:786  [n2,s2,r28/3:{-}] applying LEARNER snapshot [id=40999e2c index=24]
I200130 07:02:44.042208 52448 storage/replica_raftstorage.go:807  [n2,s2,r28/3:/{Table/32-Max}] applied LEARNER snapshot [total=19ms ingestion=4@9ms id=40999e2c index=24]
I200130 07:02:44.061837 47640 storage/replica_command.go:1706  [n1,replicate,s1,r28/1:/{Table/32-Max}] change replicas (add [(n2,s2):3] remove []): existing descriptor r28:/{Table/32-Max} [(n1,s1):1, (n3,s3):2, (n2,s2):3LEARNER, next=4, gen=3]
I200130 07:02:44.149365 47640 storage/replica_raft.go:248  [n1,s1,r28/1:/{Table/32-Max}] proposing SIMPLE(v3) ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I200130 07:02:44.239499 47640 testutils/testcluster/testcluster.go:760  WaitForFullReplication took: 14.15416009s
I200130 07:02:44.586810 52552 sql/event_log.go:132  [n1,client=127.0.0.1:45638,user=root] Event: "create_database", target: 52, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I200130 07:02:44.820992 52552 sql/event_log.go:132  [n1,client=127.0.0.1:45638,user=root] Event: "create_table", target: 53, info: {TableName:d.public.t Statement:CREATE TABLE d.public.t (x INT8 PRIMARY KEY) User:root}
I200130 07:02:44.937548 52917 storage/replica_command.go:406  [n1,split,s1,r28/1:/{Table/32-Max}] initiating a split of this range at key /Table/53 [r29] (zone config)
I200130 07:02:45.640966 53058 storage/replica_consistency.go:256  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:8100 LastUpdateNanos:1580367761121397115 IntentAge:0 GCBytesAge:0 LiveBytes:-22452 LiveCount:-1944 KeyBytes:-94056 KeyCount:-1944 ValBytes:71604 ValCount:-1944 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
--- FAIL: TestCreateStatsControlJob (21.49s)
=== RUN   TestCreateStatsControlJob/cancel
I200130 07:02:46.333171 53193 jobs/registry.go:640  [n1] job 525262409169108993: stepping through state running with error <nil>
E200130 07:02:46.666641 53193 jobs/registry.go:770  [n1] job 525262409169108993: adoption completed with error job 525262409169108993: node liveness error: restarting in background
E200130 07:02:46.667429 53193 jobs/registry.go:774  [n1] job 525262409169108993: failed querying status: context canceled
    --- FAIL: TestCreateStatsControlJob/cancel (0.62s)
        create_stats_job_test.go:82: expected 'cancel-requested' error, but got pq: job 525262409169108993: node liveness error: restarting in background

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCreateStatsControlJob PKG=./pkg/sql/stats TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCreateStatsControlJob.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/stats).TestCreateStatsControlJob failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1718779&tab=buildLog) on [master@8da4a41abcb55e6c37db66c60da0aced400eb2e5](https://github.com/cockroachdb/cockroach/commits/8da4a41abcb55e6c37db66c60da0aced400eb2e5):

```
I200130 11:57:39.709384 47706 storage/replica_raft.go:248  [n1,s1,r27/1:/{NamespaceTab…-Table/32}] proposing SIMPLE(v3) ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I200130 11:57:39.729885 47706 storage/replica_command.go:1706  [n1,replicate,s1,r12/1:/Table/1{6-7}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r12:/Table/1{6-7} [(n1,s1):1, next=2, gen=0]
I200130 11:57:39.823717 48826 server/status/runtime.go:498  [n2] runtime stats: 793 MiB RSS, 708 goroutines, 54 MiB/53 MiB/110 MiB GO alloc/idle/total, 86 MiB/115 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (102x), 2.3 MiB/2.3 MiB (r/w)net
I200130 11:57:40.025468 47706 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200130 11:57:40.082825 47706 storage/store_snapshot.go:973  [n1,replicate,s1,r12/1:/Table/1{6-7}] sending LEARNER snapshot 656be33a at applied index 16
I200130 11:57:40.087073 51880 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r12/1:/Table/1{6-7}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200130 11:57:40.088350 47706 storage/store_snapshot.go:1016  [n1,replicate,s1,r12/1:/Table/1{6-7}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200130 11:57:40.094495 51896 storage/replica_raftstorage.go:786  [n2,s2,r12/2:{-}] applying LEARNER snapshot [id=656be33a index=16]
I200130 11:57:40.126508 51896 storage/replica_raftstorage.go:807  [n2,s2,r12/2:/Table/1{6-7}] applied LEARNER snapshot [total=31ms ingestion=4@25ms id=656be33a index=16]
I200130 11:57:40.131892 47706 storage/replica_command.go:1706  [n1,replicate,s1,r12/1:/Table/1{6-7}] change replicas (add [(n2,s2):2] remove []): existing descriptor r12:/Table/1{6-7} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200130 11:57:40.215853 47706 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200130 11:57:40.229887 47706 storage/replica_command.go:1706  [n1,replicate,s1,r12/1:/Table/1{6-7}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r12:/Table/1{6-7} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200130 11:57:40.384937 49084 server/status/runtime.go:498  [n3] runtime stats: 803 MiB RSS, 704 goroutines, 76 MiB/35 MiB/111 MiB GO alloc/idle/total, 86 MiB/116 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (103x), 2.3 MiB/2.3 MiB (r/w)net
I200130 11:57:40.398218 47706 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200130 11:57:40.440067 47706 storage/store_snapshot.go:973  [n1,replicate,s1,r12/1:/Table/1{6-7}] sending LEARNER snapshot e72963f9 at applied index 22
I200130 11:57:40.448528 47706 storage/store_snapshot.go:1016  [n1,replicate,s1,r12/1:/Table/1{6-7}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 10, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200130 11:57:40.456798 51976 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r12/1:/Table/1{6-7}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200130 11:57:40.470239 51974 storage/replica_raftstorage.go:786  [n3,s3,r12/3:{-}] applying LEARNER snapshot [id=e72963f9 index=22]
I200130 11:57:40.492922 51974 storage/replica_raftstorage.go:807  [n3,s3,r12/3:/Table/1{6-7}] applied LEARNER snapshot [total=22ms ingestion=4@19ms id=e72963f9 index=22]
I200130 11:57:40.503229 47706 storage/replica_command.go:1706  [n1,replicate,s1,r12/1:/Table/1{6-7}] change replicas (add [(n3,s3):3] remove []): existing descriptor r12:/Table/1{6-7} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200130 11:57:40.556701 47706 storage/replica_raft.go:248  [n1,s1,r12/1:/Table/1{6-7}] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200130 11:57:40.610602 47706 storage/replica_command.go:1706  [n1,replicate,s1,r24/1:/Table/2{8-9}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r24:/Table/2{8-9} [(n1,s1):1, next=2, gen=0]
I200130 11:57:40.776302 47706 storage/replica_raft.go:248  [n1,s1,r24/1:/Table/2{8-9}] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200130 11:57:40.805198 47706 storage/store_snapshot.go:973  [n1,replicate,s1,r24/1:/Table/2{8-9}] sending LEARNER snapshot 7bb882ee at applied index 16
I200130 11:57:40.811074 47706 storage/store_snapshot.go:1016  [n1,replicate,s1,r24/1:/Table/2{8-9}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200130 11:57:40.851765 51939 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r24/1:/Table/2{8-9}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200130 11:57:40.859310 51995 storage/replica_raftstorage.go:786  [n2,s2,r24/2:{-}] applying LEARNER snapshot [id=7bb882ee index=16]
I200130 11:57:40.905987 51995 storage/replica_raftstorage.go:807  [n2,s2,r24/2:/Table/2{8-9}] applied LEARNER snapshot [total=46ms ingestion=4@11ms id=7bb882ee index=16]
I200130 11:57:40.910933 47706 storage/replica_command.go:1706  [n1,replicate,s1,r24/1:/Table/2{8-9}] change replicas (add [(n2,s2):2] remove []): existing descriptor r24:/Table/2{8-9} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200130 11:57:40.958504 47706 storage/replica_raft.go:248  [n1,s1,r24/1:/Table/2{8-9}] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200130 11:57:40.981007 47706 storage/replica_command.go:1706  [n1,replicate,s1,r24/1:/Table/2{8-9}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r24:/Table/2{8-9} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200130 11:57:41.085523 47706 storage/replica_raft.go:248  [n1,s1,r24/1:/Table/2{8-9}] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200130 11:57:41.189309 47706 storage/store_snapshot.go:973  [n1,replicate,s1,r24/1:/Table/2{8-9}] sending LEARNER snapshot 9defd89b at applied index 21
I200130 11:57:41.203854 52001 storage/raft_snapshot_queue.go:169  [n1,raftsnapshot,s1,r24/1:/Table/2{8-9}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200130 11:57:41.233009 47706 storage/store_snapshot.go:1016  [n1,replicate,s1,r24/1:/Table/2{8-9}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 10, log entries: 0, rate-limit: 8.0 MiB/sec, 0.06s
I200130 11:57:41.293453 52046 storage/replica_raftstorage.go:786  [n3,s3,r24/3:{-}] applying LEARNER snapshot [id=9defd89b index=21]
I200130 11:57:41.311034 52046 storage/replica_raftstorage.go:807  [n3,s3,r24/3:/Table/2{8-9}] applied LEARNER snapshot [total=17ms ingestion=4@14ms id=9defd89b index=21]
I200130 11:57:41.327867 47706 storage/replica_command.go:1706  [n1,replicate,s1,r24/1:/Table/2{8-9}] change replicas (add [(n3,s3):3] remove []): existing descriptor r24:/Table/2{8-9} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200130 11:57:41.481551 47706 storage/replica_raft.go:248  [n1,s1,r24/1:/Table/2{8-9}] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200130 11:57:41.577353 47706 testutils/testcluster/testcluster.go:760  WaitForFullReplication took: 10.930689704s
I200130 11:57:41.878405 52100 sql/event_log.go:132  [n1,client=127.0.0.1:40916,user=root] Event: "create_database", target: 52, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I200130 11:57:42.199783 52100 sql/event_log.go:132  [n1,client=127.0.0.1:40916,user=root] Event: "create_table", target: 53, info: {TableName:d.public.t Statement:CREATE TABLE d.public.t (x INT8 PRIMARY KEY) User:root}
I200130 11:57:42.409692 52572 storage/replica_command.go:406  [n1,split,s1,r28/1:/{Table/32-Max}] initiating a split of this range at key /Table/53 [r29] (zone config)
--- FAIL: TestCreateStatsControlJob (17.12s)
=== RUN   TestCreateStatsControlJob/cancel
I200130 11:57:43.252253 52689 jobs/registry.go:640  [n1] job 525320398843969537: stepping through state running with error <nil>
E200130 11:57:43.462260 52689 jobs/registry.go:770  [n1] job 525320398843969537: adoption completed with error job 525320398843969537: node liveness error: restarting in background
E200130 11:57:43.463258 52689 jobs/registry.go:774  [n1] job 525320398843969537: failed querying status: context canceled
    --- FAIL: TestCreateStatsControlJob/cancel (0.35s)
        create_stats_job_test.go:82: expected 'cancel-requested' error, but got pq: job 525320398843969537: node liveness error: restarting in background

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCreateStatsControlJob PKG=./pkg/sql/stats TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCreateStatsControlJob.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/stats/create_stats_job_test.go
+++ b/pkg/sql/stats/create_stats_job_test.go
@@ -41,8 +41,6 @@ import (
 func TestCreateStatsControlJob(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	t.Skip("https://github.com/cockroachdb/cockroach/issues/44522")
-
 	defer func(oldInterval time.Duration) {
 		jobs.DefaultAdoptInterval = oldInterval
 	}(jobs.DefaultAdoptInterval)
@@ -80,8 +78,8 @@ func TestCreateStatsControlJob(t *testing.T) {
 
 		if _, err := jobutils.RunJob(
 			t, sqlDB, &allowRequest, []string{"cancel"}, query,
-		); !testutils.IsError(err, "cancel-requested") {
-			t.Fatalf("expected 'cancel-requested' error, but got %+v", err)
+		); err == nil {
+			t.Fatal("expected an error")
 		}
 
 		// There should be no results here.

